common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -181,6 +182,7 @@ def publish_event(event_name, **kwargs):
         if overlimit_block_keys:
             # Publish an event for analytics purposes:
             publish_event("removed", removed=format_block_keys(overlimit_block_keys), reason="overlimit")
+            previous_count = len(selected)
```

Comment:
Could be worth also updating the `previous_count` variable after the last event below ("assigned", line 201), to not forget to update the count if additional code/events were added afterwards, later on.